### PR TITLE
atlas: preparations for referenced column types

### PR DESCRIPTION
**Pull request by masseelch**
This PR introdues first steps to enable referenced column types. This is needed for the postgres implementation of enums.

The goal is to support HCL like the following:

```hcl
schema "sch" {}
table "tbl" {
  column "col" {
    type = enum.my_enum
  }
}
enum "my_enum" {
  values = ["on", "off"]
}
```

The first approach was to change `sqlspec.Table.Colum` from `*schemaspec.Type` to `schemaspec.Value` but handling a `Value` required different handling on `sqlspec.Table.Column.Type` and `sqlspec.Table.Column.Default` and the current structure didn't allow for this.

The current approach is valid too and all the implementation details are hidden from the actual `schema.Schema` the atlas internals work with.

Please note, that implementation for converting between `sqlspec.Schema` and `schema.Schema` will follow in another PR.

### CODE REVIEW

File: sql/mysql/sqlspec_test.go [hunk 1]
```diff
@@ -660,10 +660,6 @@ func TestTypes(t *testing.T) {
 			typeExpr: "longtext",
 			expected: &schema.StringType{T: TypeLongText},
 		},
-		{
-			typeExpr: `enum("a","b")`,
-			expected: &schema.EnumType{T: "enum", Values: []string{"a", "b"}},
-		},
```
[the remark this answers is not in the document]
Comment (masseelch) [reply sql/mysql/sqlspec_test.go:666]: https://github.com/ariga/atlas/blob/e2d7dad198ad537e6045e37ea13f6924c1f893a5/sql/mysql/sqlspec_test.go#L528

Was a duplicate test. There still is one.

File: sql/sqlspec/sqlspec.go [hunk 2]
```diff
@@ -25,6 +25,14 @@ type (
...
 	}
 
+	// Enum holds a specification for an enum, that can be referenced as a column type.
+	Enum struct {
+		Name   string          `spec:",name"`
+		Schema *schemaspec.Ref `spec:"schema"`
+		Values []string        `spec:"values"`
+		schemaspec.DefaultExtension
+	}
+
```
Comment (rotemtam) [inline sql/sqlspec/sqlspec.go:35]: shouldn't this be in the postgres package?

File: sql/internal/specutil/spec.go [hunk 3]
```diff
@@ -48,6 +48,7 @@ func ListAttr(k string, litValues ...string) *schemaspec.Attr {
 type doc struct {
 	Tables  []*sqlspec.Table  `spec:"table"`
 	Schemas []*sqlspec.Schema `spec:"schema"`
+	Enums   []*sqlspec.Enum   `spec:"enum"`
```
Comment (rotemtam) [inline sql/internal/specutil/spec.go:51]: should be specific to postgres imo

**Comment by masseelch** [comment]
@rotemtam moved it into `postgres`. Please have a look again.

**Review by rotemtam — APPROVED** [review]

**Pull request merged**